modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/loader/parser/java/JavaExtensionModelParserTestCase.java

```diff
@@ -44,6 +59,22 @@ public class JavaExtensionModelParserTestCase {
   @Rule
   public ExpectedException expectedException = ExpectedException.none();
 
+  @Test
+  @Issue("W-12622240")
+  @Description("Verify that ExtensionModel for an Extension with java data types such as LocalDateTime, loads the correct MetadataType i.e. DateTimeType and works in Java 17")
+  public void getParameterizedWithJavaFieldsExtensionUsingSdkApi() {
+    ExtensionModel extensionModel = loadExtension(ParameterizedWithJavaTypeExtension.class);
+    ConfigurationModel configModel = extensionModel.getConfigurationModels().get(0);
+    ParameterModel parameterModel = configModel.getAllParameterModels().get(0);
+    DefaultObjectType objectType = (DefaultObjectType) parameterModel.getType();
+    assertThat(getTypeId(objectType).get(),
+               is("org.mule.runtime.module.extension.internal.loader.parser.java.JavaExtensionModelParserTestCase.SimplePojoWithTime"));
+
+    List<ObjectFieldType> objectFieldTypes = objectType.getFields().stream().collect(toList());
+    assertThat(objectFieldTypes.get(2).getKey().getName().getLocalPart(), is("dateTime"));
```

Comment:
get the field by name instead of relying on the index, makes the test more clear and prevents flakyness.